### ISSUE TITLE
Fix S2583 FP: Should not raise on `double` condition

### ISSUE THREAD

**Issue by nrankin18** [SonarSource/sonar-dotnet#8470]
### Description

Rule S2583 does not correctly identify all possible values of a double and incorrectly assumes a double cannot fall between (0, 1).

### Repro steps

		
```cs
double t = 0.5;
if (t <= 0) {
	return "a";
}

if (t >= 1) {  // S2583 is flagged. Assumes t must be >= 1
	return "b";
}

return "c";
```

### Expected behavior

The issue should not be flagged since t can be (0, 1).

### Actual behavior

S2583 is flagged.

### Known workarounds

Please provide a description of any known workarounds.

### Related information

* C# version: 10.0
* Visual Studio version: 17.8.2
* dotnet version : net6.0
* Operating System: Windows 11 Home 10.0.22000 Build 22000

**Comment by pavel-mikula-sonarsource** [comment]
Hi,

Thank you for reporting this case, I confirm it as FP. It's an interesting one, as we should not be tracking `double` type here.
